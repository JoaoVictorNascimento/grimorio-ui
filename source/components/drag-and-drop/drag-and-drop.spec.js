import '../../../internals/test/helper';

import DragAndDrop from './drag-and-drop-component';

/** @test {DragAndDrop} */
describe('DragAndDrop component', () => {
/** @test {DragAndDrop#render} */
  describe('#render', () => {
    it('render correctly', () => {
      const initialData = {
        tasks: {
          'task-1': { id: 'task-1', content: 'Teste 1' },
          'task-2': { id: 'task-2', content: 'Teste 2' },
          'task-3': { id: 'task-3', content: 'Teste 3' },
          'task-4': { id: 'task-4', content: 'Teste 4' },
        },
        columns: {
          'column-1': {
            id: 'column-1',
            title: 'Drag and Drop',
            taskIds: ['task-1', 'task-2', 'task-3', 'task-4'],
          },
        },
        columnOrder: ['column-1'],
      };

      const wrapper = shallow(
        <DragAndDrop initialData={initialData} />
      );
      expect(wrapper.length).toEqual(1);
    });
  });
});
