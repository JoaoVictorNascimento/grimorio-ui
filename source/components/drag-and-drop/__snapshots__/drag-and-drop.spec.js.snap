// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DragAndDrop component #render render correctly 1`] = `
"<DragDropContext onDragEnd={[Function]}>
  <[object Object] droppableId=\\"droppable-default\\" mode=\\"standard\\" type=\\"DEFAULT\\" direction=\\"vertical\\" isDropDisabled={false} isCombineEnabled={false} ignoreContainerClipping={false} renderClone={{...}} getContainerForClone={[Function: getBody]}>
    <undefined />
  </[object Object]>
</DragDropContext>"
`;
